main.cpp

```diff
@@ -168,108 +192,202 @@ struct OvenStove
+        std::cout << "Season with " + mainCondiment << std::endl;
+        isSeasoned = true;
+    }
+
+}
+
+void OvenStove::FoodItem::cut(int numberOfPortion, std::string tool)
+{
+    if (expirationDayRemain <= 0)
+    {
+        std::cout << "Expired food!" << std::endl;
+        return;
+    }
+
+    if (numberOfPortion > 1)
+    {
+        std::cout << "Cut the food into " 
+                + std::to_string(numberOfPortion) 
+                + " with " 
+                + tool 
+                << std::endl;
+    }
+    else
+    {
+        std::cout << "Do nothing" << std::endl;
+    }
+}
+
+void OvenStove::broilFood(OvenStove::FoodItem food, int temperature, int durationInMinute)
+{
+    std::cout << "Broiling " + food.name 
+                    + " at " + std::to_string(temperature) + "F"
+                    + " in " + std::to_string(durationInMinute) + "minutes";
+}
+
+void bakeFood(OvenStove::FoodItem food, int temperature, int durationInMinute, int rackId, bool preHeat)
+{
+    if (rackId < 0)
+    {
+        std::cout << "ERROR!" << std::endl;
+        return;
+    }
+
+    if (preHeat == false)
+    {
+        std::cout << "Please pre heat oven before start!" << std::endl;
+        return;
+    }
+
+    std::cout << "Baking " + food.name 
+                    + " at " + std::to_string(temperature) + "F"
+                    + " in " + std::to_string(durationInMinute) + "minutes";
+}
+
+void lightUp(bool lightStatus)
+{
+    if (lightStatus == false)
+    {
+        std::cout << "Light ON" << std::endl;
+        lightStatus = true;
+    }
+    else
+    {
+        std::cout << "Do NOTHING" << std::endl;
+    }
+}
+
 struct Laptop
 {
-    // model name
     std::string modelName = "Thinkpad";
-    // screen display size in inch
     float screenDisplay = 15.6f;
-    // RAM size in GB
     int RAMSize = 32;
-    // storage size in GB
     int storageSize = 2000;
-    // weight in lbs
     float weight = 6.2f;
 
-    // connect to Wifi
     void connectToWifi(bool connectionStatus = false);
-    // check available OS update
     std::string checkAvailableOSUpdate(int currentOSId);
-    // play audio
     void playAudio(int audioFileId = 0, int volume = 50, std::string playMode = "Headphones");
 };
 
-/*
-Thing 3) fruit tree
-5 properties:
-    1) name (std::string)
-    2) height (float)
-    3) bark thickness diameter (double)
-    4) fruit color (std::string)
-    5) harvest season (std::string)
-3 things it can do:
-    1) produce oxygen
-    2) absorb carbon dioxide
-    3) produce fruit
- */
+void Laptop::connectToWifi(bool connectionStatus)
+{
+    if (connectionStatus == false)
+    {
+        connectionStatus = true;
+    }
+    std::cout << "You are connected to the Internet" << std::endl;
+}
+
+std::string Laptop::checkAvailableOSUpdate(int currentOSId)
+{
+    if (currentOSId < 10) // assume 10 is the latest OS id
+    {
+        return "Update version 10.";
+    }
+    else
+    {
+        return "Up to date!";
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise